src/Compilers/CSharp/Portable/Symbols/Source/SourceLocalSymbol.cs

```diff
@@ -597,40 +603,35 @@ private class PossiblyImplicitlyTypedOutVarLocalSymbol : SourceLocalSymbol
                                  binder.ScopeDesignator == declarationExpression.Parent.Parent :
                                  binder.ScopeDesignator.Contains(declarationExpression.Parent.Parent.Parent));
 #endif
+                _containingInvocation = containingInvocation;
             }
 
+            internal override SyntaxNode ForbiddenZone => _containingInvocation;
+
+            internal override ErrorCode ForbiddenDiagnostic => ErrorCode.ERR_ImplicitlyTypedOutVariableUsedInTheSameArgumentList;
```

Comment:
Couldn't we use a single generic message for "Can't use a variable in the code that initializes it"?